lib/puppet/provider/service/systemd.rb

```diff
@@ -145,5 +157,12 @@ def stopcmd
   def statuscmd
     [command(:systemctl), "is-active", @resource[:name]]
   end
+
+  def prepare_error_message(name, action)
+    error_return = "Systemd #{action} for #{name} failed:"
+    journalctl_output = execute("journalctl -n 50 --since '5 minutes ago' -u #{name} --no-pager")
```

Comment:
If this command fails you most likely have much bigger problems, and there's a good chance that Puppet will crash and burn entirely shortly. But if this does happen and throws an exception, you'll most likely end up here: https://github.com/puppetlabs/puppet/blob/master/lib/puppet/transaction/resource_harness.rb#L26